.travis.yml

```diff
@@ -3,6 +3,15 @@ node_js:
   - '8'
   - '6'
   - '4'
+matrix:
+  include:
+    - node_js: 8
+      env: INTEGRATION=true
+  allow_failures:
+    - node_js: 8
+      env: INTEGRATION=true
 before_install:
   - npm i -g npm
+script:
+  - if [ "$INTEGRATION" = true ]; then npm run integration else npm test; fi
```

Comment:
What do you mean with `quote the commands`? I can't do this:

```
- if [ "$INTEGRATION" = true ]; then "npm run integration" else "npm test"; fi
```

I can do this if you like it more

```
- if [ "$INTEGRATION" = true ]; then npm run integration; fi
- if [ -z "$INTEGRATION ]; then npm test; fi
```

Not sure about the `-z`, have to verify. But you'll get the idea :).